subprojects/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/ArtifactRepositoryPluginResolver.java

```diff
@@ -39,18 +39,22 @@ public ArtifactRepositoryPluginResolver(String name, DependencyResolutionService
     }
 
     @Override
-    public void resolve(final PluginRequest pluginRequest, PluginResolutionResult result) throws InvalidPluginRequestException {
-        if (pluginRequest.getVersion() == null) {
+    public void resolve(final InternalPluginRequest pluginRequest, PluginResolutionResult result) throws InvalidPluginRequestException {
+        String version = pluginRequest.getVersion();
+        if (version == null) {
             result.notFound(name, "plugin dependency must include a version number for this source");
             return;
         }
-        if (pluginRequest.getVersion().endsWith("-SNAPSHOT")) {
-            result.notFound(name, "snapshot plugin versions are not supported");
-            return;
-        }
-        if (versionSelectorScheme.parseSelector(pluginRequest.getVersion()).isDynamic()) {
-            result.notFound(name, "dynamic plugin versions are not supported");
-            return;
+
+        if (null == pluginRequest.getArtifact()) {
```

Comment:
The two checks below should always be done. Either on the plugin version or on the version on the `ModuleVersionSelector`. We don't support dynamic/changing versions in either case here.